Xamarin.Forms.Core/BindableObject.cs

```diff
@@ -194,7 +194,12 @@ public static void SetInheritedBindingContext(BindableObject bindable, object va
 
 		protected void ApplyBindings() => ApplyBindings(skipBindingContext: false, fromBindingContextChanged: false);
 
-		protected virtual void OnBindingContextChanged() => BindingContextChanged?.Invoke(this, EventArgs.Empty);
+		protected virtual void OnBindingContextChanged()
+		{
+			BindingContextChanged?.Invoke(this, EventArgs.Empty);
+			if (Shell.GetBackButtonBehavior(this) is BackButtonBehavior buttonBehavior)
```

Comment:
Should we generalize this into a method on Shell? There's a few things it seems like this should apply to?

`Shell.PropagateBindingContext`

Then it can do this same thing here to
`SetTitleView`
`SetSearchHandler`

I think that's the rest for now

Then it can also check if it's even running inside a shell before trying to propagate any of these values at all.

Reply:
@PureWeen this is the first time we're declaring attached BP of type BO, and I choose not to generalize for now to see:
- if it's needed,
- what pattern emerges from the use cases

As I foresee it, the generalization will probably happen in BP/BO, and not shell, making sure the context is propagated when a property is added (or attached), and when the BC changes. This is not a change I'd like to see in 4.0.0, so I think this should be left as-is.

We shouldn't check if it's running inside the Shell for the following reasons:
- checking if a Page is inside a Shell is more expensive than propagating the BC
- applying the BC when a page is parented to a Shell is a nightmare
- SearchHandler, e.g., could be reused outside the context of a Shell (provided the renderers are implemented)
- this would break expectations
- we still hope all app will end up being Shell-based, so no need to check :P

Reply:
Sounds good. SetTitleView needs the same treatment right? and I think that'll be the last one.

Loop that in with this PR? or make a new one?